### ISSUE TITLE
Calendar grid in edit field does not work correctly in Firefox

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#9172]
### Steps to reproduce:
1. Open this website in Firefox
https://www.flysafair.co.za/
2. Activate "round trip"
3. With NVDA  in Focus mode, tab to the "departure or to "return"
3. Those edit fields are read only but you can change the value by pressing the arrow keys (left and right change the days, up and down change the weeks)
4. After changing the value in "departure", change the date in return

### Actual behavior:
NVDA is changing the departure date, but NVDA reports blank, when changing day by day with left and right arrow. When focus is in the return date field, the return date cannot be changed. In some cases, changing the return date results in a changed departure date.

### Expected behavior:
NVDA should report the days when changing day by day with left and right arrow. When changing the return date, if possible, the departure date should remain as it was selected before.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
2018.4.1

#### Windows version:
Windows 10 1809 Update

#### Name and version of other software in use when reproducing the issue:
Firefox 64.0.2

#### Does the issue still occur after restarting your PC?
yes

#### Have you tried any other versions of NVDA?
no

**Comment by Brian1Gaff** [comment]
I've had awful trouble with the  dates moving back a week when I try to 
move them on as well. Not tried other browsers other than waterfox which 
does the same sort of funny stuff. (that is a techical term meaning, ehat 
the xxx is happening)
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
----- Original Message -----

**Comment by Adriani90** [comment]
cc: @jcsteh

**Comment by jcsteh** [comment]
I tried this in both Firefox and Chrome. In Firefox, the date doesn't change as it should. However, this isn't accessibility specific. In Chrome, the date does change, but changes aren't read after the first press. Using a text entry field for a calendar like this isn't going to be accessible.

Certainly, the accessibility issues here are problems with the site. It's possible the problems with the arrow keys in Firefox are also a problem with the site; e.g. they might be using non-standard code which only works in specific browsers.

**Comment by Adriani90** [comment]
Thanks for the info Jamie. Is this something which could be improved via displaymodel? I mean is there any possibility to track such inaccessible edit fields via GDI calls or so? It's just for my understanding. I don't think this would be viable, probably it would have also negative impacts on performance and stability if possible at all.

**Comment by jcsteh** [comment]
Display model can't be used in web browsers at all. But it wouldn't help anyway.

The information you want is available; it just can't be reported automatically because they're using a text box in a weird way. However, if you press read line (NVDA+upArrow/NVDA+l) or report focus (NVDA+tab), the changed date is reported just fine.

Btw, the arrow keys now seem to change the date as expected in Firefox.

**Comment by Adriani90** [comment]
Yes this seems partly improved now. I am closing this anyway since there is a good workaround for getting the current value and probably a can't fix anyway. Thanks for the info.
